core/src/toga/widgets/base.py

```diff
@@ -33,6 +54,20 @@ def __init__(
         :param style: A style object. If no style is provided, a default style
             will be applied to the widget.
         """
+        # if the object has _USE_DEBUG_BACKGROUND=True and layout debug mode is on, change bg color
+        if hasattr(self, "_USE_DEBUG_BACKGROUND") and self._USE_DEBUG_BACKGROUND:
```

Comment:
Hope I'm not butting in here, but a handy thing you may not be aware of: [`getattr()`](https://docs.python.org/3/library/functions.html#getattr) can take an optional third parameter as a default to supply if the attribute is missing. Dictionaries have a roughly analogous [`get()`](https://docs.python.org/3/library/stdtypes.html#dict.get) method, which returns `None` or a user-supplied default if the key is missing.

So lines 58 and 59 could do the same thing and be written as:

```python
if getattr(self, "_USE_DEBUG_BACKGROUND", False):
    if environ.get("TOGA_DEBUG_LAYOUT") == "1":
```